cpp/include/typedb/answer/conceptmap.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#pragma once
+
+#include "typedb/common/native.hpp"
+#include "typedb/common/iterator.hpp"
+#include "typedb/concept/concept.hpp"
+
+namespace TypeDB {
+
+using ConceptIterable = TypeDBIterable<_native::ConceptIterator, _native::Concept, TypeDB::Concept>;
+using ConceptIterator = TypeDBIterator<_native::ConceptIterator, _native::Concept, TypeDB::Concept>;
+
+class ConceptMap {
+    
+    friend class TypeDBIterator<_native::ConceptMapIterator, _native::ConceptMap, ConceptMap>;
+
+   private: 
+
+    NativePointer<_native::ConceptMap> conceptMapNative;
+    
+   public:
+    ConceptMap(_native::ConceptMap*);
+    ConceptMap(const ConceptMap&) = delete;
+    ConceptMap(ConceptMap&&);
+    
+    ConceptMap& operator=(const ConceptMap&) = delete;
+    ConceptMap& operator=(ConceptMap&&);
+
+    StringIterable variables();
+    ConceptIterable concepts(); // TODO: Are we likely to create multiple copies to the same native object if we expose this? Should we copy the whole map over?
```

Comment:
Specific question:
If I iterate through `concepts()` and get a variable through get, Are they the same rust object?
Are these ever the same object? (things like DatabaseManager? )

General question: 
If I call the same method twice, will I get a different native pointer each time?

E.g.: 
If I have an attribute and get its value ?